service-base/src/test/java/org/eclipse/hono/service/metric/MicrometerBasedMetricsTest.java

```diff
@@ -157,4 +159,27 @@ public void testReportTelemetryInvokesLegacyMetrics(final MeterRegistry registry
 
         verify(legacyMetrics).incrementProcessedMessages(eq(EndpointType.TELEMETRY), eq("tenant"));
     }
+
+    /**
+     * Verifies that the calculated payload size for messages of different sizes are correct.
+     * 
+     * @param registry : the registry that the tests should be run against.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testCalculatePayloadSize(final MeterRegistry registry) {
```

Comment:
I'd rather have a test which verifies that the correct values are set on the MeterRegistry when reporting messages via  Metrics' interface methods ..

Reply:
@sophokles73 I have created tests accordingly and pushed the changes.